financialaid/api_test.py

```diff
@@ -187,3 +214,63 @@ def test_determine_auto_approval(self):  # pylint: disable=no-self-use
             country_of_income="KP"
         )
         assert determine_auto_approval(financial_aid) is True
+
+    def test_get_no_discount_tier_program(self):
+        """
+        Tests get_no_discount_tier_program()
+        """
+        # 100k tier program is the one with no discount
+        assert get_no_discount_tier_program(self.program.id).id == self.tier_programs["100k"].id
+
+    def test_get_course_price_for_learner(self):
+        """
+        Tests get_course_price_for_learner()
+        """
+        # Enrolled and has approved financial aid
+        expected_response = {
+            "has_financial_aid_request": True,
+            "course_price": self.course_price.price - self.financialaid_approved.tier_program.discount_amount,
+            "financial_aid_adjustment": True,
+            "financial_aid_availability": True
+        }
+        self.assertDictEqual(
+            get_course_price_for_learner(self.enrolled_profile.user, self.program),
+            expected_response
+        )
+        # Enrolled and has pending financial aid
+        expected_response = {
+            "has_financial_aid_request": True,
+            "course_price": self.course_price.price,
+            "financial_aid_adjustment": False,
+            "financial_aid_availability": True
+        }
+        self.assertDictEqual(
+            get_course_price_for_learner(self.enrolled_profile2.user, self.program),
+            expected_response
+        )
+        # Enrolled and has no financial aid
+        expected_response = {
+            "has_financial_aid_request": False,
+            "course_price": self.course_price.price,
+            "financial_aid_adjustment": False,
+            "financial_aid_availability": True
+        }
+        self.assertDictEqual(
+            get_course_price_for_learner(self.enrolled_profile3.user, self.program),
+            expected_response
+        )
+        # Enrolled but program has no financial aid availability
+        self.program.financial_aid_availability = False
```

Comment:
Please split up these tests into separate cases, one per case being tested (basically, each test should end with an assert being made)